src/Microsoft.Graph.Core/Requests/Content/BatchRequestContent.cs

```diff
@@ -0,0 +1,198 @@
+                });
+
+            this.Headers.Add("Content-Type", "application/json");
+
+            if (batchRequestSteps.Count() > MAX_NUMBER_OF_REQUESTS)
+                throw new ServiceException(new Error {
+                    Code = ErrorConstants.Codes.MaximumValueExceeded,
+                    Message = string.Format(ErrorConstants.Messages.MaximumValueExceeded, "Number of batch request steps", MAX_NUMBER_OF_REQUESTS)
+                });
+
+            BatchRequestSteps = new Dictionary<string, BatchRequestStep>();
+            foreach (BatchRequestStep requestStep in batchRequestSteps)
+                AddBatchRequestStep(requestStep);
+        }
+
+        /// <summary>
+        /// Adds a <see cref="BatchRequestStep"/> to batch request content if doesn't exists.
+        /// </summary>
+        /// <param name="batchRequestStep">A <see cref="BatchRequestStep"/> to add.</param>
+        /// <returns>True or false based on addition or not addition of the provided <see cref="BatchRequestStep"/>. </returns>
+        public bool AddBatchRequestStep(BatchRequestStep batchRequestStep)
+        {
+            if (batchRequestStep == null || BatchRequestSteps.ContainsKey(batchRequestStep.RequestId))
+                return false;
+            BatchRequestSteps.Add(batchRequestStep.RequestId, batchRequestStep);
+            return true;
+        }
+
+        /// <summary>
+        /// Removes a <see cref="BatchRequestStep"/> from batch request content for the specified id.
+        /// </summary>
+        /// <param name="requestId">A batch request id to remove.</param>
+        /// <returns>True or false based on removal or not removal of a <see cref="BatchRequestStep"/>.</returns>
+        public bool RemoveBatchRequestStepWithId(string requestId)
+        {
+            bool isRemoved = false;
+            if (BatchRequestSteps.ContainsKey(requestId)) {
+                BatchRequestSteps.Remove(requestId);
+                isRemoved = true;
+                foreach (KeyValuePair<string, BatchRequestStep> batchRequestStep in BatchRequestSteps)
+                {
+                    if (batchRequestStep.Value != null && batchRequestStep.Value.DependsOn != null)
+                        while (batchRequestStep.Value.DependsOn.Remove(requestId)) ;
+                }
+            }
+            return isRemoved;
+        }
+
+        internal async Task<JObject> GetBatchRequestContentAsync()
+        {
+            JObject batchRequest = new JObject();
+            JArray batchRequestItems = new JArray();
+
+            foreach (KeyValuePair<string, BatchRequestStep> batchRequestStep in BatchRequestSteps)
+                batchRequestItems.Add(await GetBatchRequestContentFromStepAsync(batchRequestStep.Value));
+
+            batchRequest.Add("requests", batchRequestItems);
+
+            return batchRequest;
+        }
+
+        private async Task<JObject> GetBatchRequestContentFromStepAsync(BatchRequestStep batchRequestStep)
+        {
```

Comment:
I suggest that we make BatchRequestStep a serializable object - decorate the BatchRequestStep properties with JsonPropertyAttribute. It would remove the need for this task. Optional.

Reply:
Can we perf test the difference before making that change? :-)

Reply:
@darrelmiller @MIchaelMainer  Here are my findings: 
- It takes **~30 milliseconds** to serialize a single `BatchRequestStep` object decorated with JsonPropertyAttribute using the default provider serializer.
- It takes **~0.0324 milliseconds** to serialize the same `BatchRequestStep` object by converting it to a JObject and calling .ToString on it :-).

Reply:
Good lesson.